wsmaster/che-core-api-factory/src/test/java/org/eclipse/che/api/factory/server/DefaultFactoryParameterResolverTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.factory.server;
+
+import static org.eclipse.che.api.factory.shared.Constants.URL_PARAMETER_NAME;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.eclipse.che.api.devfile.model.Tool;
+import org.eclipse.che.api.devfile.server.DevfileManager;
+import org.eclipse.che.api.devfile.server.FileContentProvider;
+import org.eclipse.che.api.devfile.server.convert.CommandConverter;
+import org.eclipse.che.api.devfile.server.convert.DevfileConverter;
+import org.eclipse.che.api.devfile.server.convert.ProjectConverter;
+import org.eclipse.che.api.devfile.server.convert.tool.ToolProvisioner;
+import org.eclipse.che.api.devfile.server.convert.tool.ToolToWorkspaceApplier;
+import org.eclipse.che.api.devfile.server.schema.DevfileSchemaProvider;
+import org.eclipse.che.api.devfile.server.validator.DevfileIntegrityValidator;
+import org.eclipse.che.api.devfile.server.validator.DevfileSchemaValidator;
+import org.eclipse.che.api.factory.server.urlfactory.URLFactoryBuilder;
+import org.eclipse.che.api.factory.server.urlfactory.URLFetcher;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+@Listeners(value = {MockitoTestNGListener.class})
+public class DefaultFactoryParameterResolverTest {
+
+  private static final String DEVFILE =
+      ""
+          + "specVersion: 0.0.1\n"
+          + "name: test\n"
+          + "tools:\n"
+          + "- type: kubernetes\n"
+          + "  name: tool\n"
+          + "  local: ../localfile\n";
+
+  private static final String LOCALFILE = "" + "type: ConfigMap";
```

Comment:
What the magic with concatenating empty string here?

Reply:
A remnant of abandoned approach to testing it. Thanks for spotting it :)